src/rtree/index.c

```diff
@@ -56,37 +56,37 @@ struct Node * Xastir_RTreeNewIndex(void)
 //
 int Xastir_RTreeSearch(struct Node *N, struct Rect *R, SearchHitCallback shcb, void* cbarg)
 {
-	register struct Node *n = N;
-	register struct Rect *r = R; // NOTE: Suspected bug was R sent in as Node* and cast to Rect* here. Fix not yet tested.
-	register int hitCount = 0;
-	register int i;
-
-	assert(n);
-	assert(n->level >= 0);
-	assert(r);
-
-	if (n->level > 0) /* this is an internal node in the tree */
-	{
-		for (i=0; i<Xastir_NODECARD; i++)
-			if (n->branch[i].child &&
-			    Xastir_RTreeOverlap(r,&n->branch[i].rect))
-			{
-				hitCount += Xastir_RTreeSearch(n->branch[i].child, R, shcb, cbarg);
-			}
-	}
-	else /* this is a leaf node */
-	{
-		for (i=0; i<Xastir_LEAFCARD; i++)
-			if (n->branch[i].child &&
-			    Xastir_RTreeOverlap(r,&n->branch[i].rect))
-			{
-				hitCount++;
-				if(shcb) // call the user-provided callback
-					if( ! shcb(n->branch[i].child, cbarg))
-						return hitCount; // callback wants to terminate search early
-			}
-	}
-	return hitCount;
+    struct Node *n = N;
+    struct Rect *r = R; // NOTE: Suspected bug was R sent in as Node* and cast to Rect* here. Fix not yet tested.
```

Comment:
I haven't dealt with the rtree stuff before so I'll let others verify functionality.

Reply:
Most of these changes are just whitespace.  The original rtree code that we just dropped into Xastir was indented using tabs, which is forbidden by our style.  As long as I was going through the code fixing very minor warnings (which can be seen by looking at the various master branch Travis builds), I fixed the indentation to use spaces of the appropriate length for this project.

Reply:
And as far as removing register keywords, this is the correct thing to do.  No modern compiler respects them anymore, and newer C++ standards have even removed the keyword altogether.  It is the warnings that come from using this keyword that this PR is addressing, and therefore removing the keyword is essential.